src/request/mod.rs

```diff
@@ -193,29 +205,38 @@ impl<B: Body> PreparedRequest<B> {
     /// Send this request and wait for the result.
     pub fn send(&mut self) -> Result<Response> {
         let mut url = self.url.clone();
-        set_host(&mut self.base_settings.headers, &url)?;
+        set_host(&mut self.base_settings.headers, &url)?; // TODO update when redirected
```

Comment:
Address this todo, should `Host` be the proxy or the remote website, to be verified.